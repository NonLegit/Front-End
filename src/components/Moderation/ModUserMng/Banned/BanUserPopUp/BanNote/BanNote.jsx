import * as React from 'react';
import { TextField } from '@mui/material';
import { StyledBox, StyledFont } from './styles';

/**
 * BanNote
 * @component
 * @property  {string} note a note from the moderator about the banned user
 * @return {React.Component} - Bannote (text field)
 */

function BanNote(props) {
  const { note } = props;
  const CHARACTER_LIMIT = 300;
  const [values, setValues] = React.useState({
    name: note,
  });
  const handleChangeCharNum = (name) => (event) => {
    setValues({ ...values, [name]: event.target.value });
  };
  return (
    <StyledBox>
      <StyledFont>MOD NOTE</StyledFont>
      <TextField
        id="banNote"
        placeholder="Mod Note"
        size="small"
        fullWidth
        inputProps={{
          maxLength: CHARACTER_LIMIT,
        }}
        value={values.name}
        helperText={`${CHARACTER_LIMIT - values.name.length} Characters remaining`}
        onChange={handleChangeCharNum('name')}
      />
    </StyledBox>
  );
}

export default BanNote;
