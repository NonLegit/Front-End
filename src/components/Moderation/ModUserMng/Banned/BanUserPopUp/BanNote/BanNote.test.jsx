import renderer from 'react-test-renderer';
import BanNote from './BanNote';

test('test snapshot', async () => {
  const tree = renderer.create(<BanNote note="spam" />).toJSON();
  expect(tree).toMatchSnapshot();
});
