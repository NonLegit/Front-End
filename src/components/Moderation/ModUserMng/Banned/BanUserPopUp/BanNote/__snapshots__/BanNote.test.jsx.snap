// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-9pcsoq"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1p94fvm-MuiTypography-root"
  >
    MOD NOTE
  </p>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-describedby="banNote-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="banNote"
        maxLength={300}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Mod Note"
        required={false}
        type="text"
        value="spam"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained MuiFormHelperText-filled css-k4qjio-MuiFormHelperText-root"
      id="banNote-helper-text"
    >
      296 Characters remaining
    </p>
  </div>
</div>
`;
